ui/src/ethereum/walletConnect.ts

```diff
@@ -210,6 +218,84 @@ function createConnector(): Connector {
   return connector;
 }
 
+// WalletConnect test client that is backed by an in-memory wallet and
+// automatically signs and submits transactions without user
+// interaction.
+export class TestClient implements WalletConnect {
+  private _connection: svelteStore.Writable<Connection | undefined> =
+    svelteStore.writable(undefined);
+  private wallet: ethers.Wallet;
+  private provider: ethers.providers.Provider;
+
+  public connection: svelteStore.Readable<Connection | undefined>;
+
+  constructor(mnemonic: string) {
+    this.connection = svelteStore.derived(this._connection, x => x);
+    this.wallet = ethers.Wallet.fromMnemonic(mnemonic);
+    this.provider = ethers.providers.InfuraProvider.getWebSocketProvider(
+      "rinkeby",
+      "de5e2a8780c04964950e73b696d1bfb1"
```

Comment:
We're using this in multiple places, can we extract this into `config.ts`?

Reply:
Why do we need the live provider actually? Shouldn't all networking be stubbed out?

Reply:
I’ve extracted the key.



> Why do we need the live provider actually? Shouldn't all networking be stubbed out?

The idea for this is to test the app with rinkeby and have a wallet that automatically signs all transactions. I’ll follow this up with a PR that expands this so we can run against a local test net.